plugins/source/azure/resources/services/network/virtual_networks.go

```diff
@@ -1,988 +1,142 @@
-								Type:        schema.TypeString,
-							},
-							{
-								Name:        "location",
-								Description: "Resource location.",
-								Type:        schema.TypeString,
-							},
-							{
-								Name:        "tags",
-								Description: "Resource tags.",
-								Type:        schema.TypeJSON,
-							},
-						},
-					},
-				},
-			},
+			virtualNetworkGateways(),
 		},
 	}
 }
 
-// ====================================================================================================================
-//
-//	Table Resolver Functions
-//
-// ====================================================================================================================
 func fetchNetworkVirtualNetworks(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan<- interface{}) error {
 	svc := meta.(*client.Client).Services().Network.VirtualNetworks
-	response, err := svc.ListAll(ctx)
-	if err != nil {
-		return diag.WrapError(err)
-	}
-	for response.NotDone() {
-		res <- response.Values()
-		if err := response.NextWithContext(ctx); err != nil {
-			return diag.WrapError(err)
-		}
-	}
-	return nil
-}
-func resolveNetworkVirtualNetworksDhcpOptionsDnsServers(ctx context.Context, meta schema.ClientMeta, resource *schema.Resource, c schema.Column) error {
-	vn := resource.Item.(network.VirtualNetwork)
-	if vn.DhcpOptions == nil || vn.DhcpOptions.DNSServers == nil || len(*vn.DhcpOptions.DNSServers) == 0 {
-		return nil
-	}
-	ips := make([]net.IP, 0, len(*vn.DhcpOptions.DNSServers))
-	for _, ip := range *vn.DhcpOptions.DNSServers {
-		i := net.ParseIP(ip)
-		if i == nil {
-			return diag.WrapError(fmt.Errorf("wrong format of IP: %s", ip))
-		}
-		ips = append(ips, i)
-		net.ParseIP(ip)
-	}
-	return diag.WrapError(resource.Set(c.Name, ips))
-}
-func resolveNetworkVirtualNetworksIpAllocations(ctx context.Context, meta schema.ClientMeta, resource *schema.Resource, c schema.Column) error {
-	vn := resource.Item.(network.VirtualNetwork)
-	if vn.IPAllocations == nil {
-		return nil
-	}
-	allocations := make([]string, 0, len(*vn.IPAllocations))
-	for _, a := range *vn.IPAllocations {
-		allocations = append(allocations, *a.ID)
-	}
-	return diag.WrapError(resource.Set(c.Name, allocations))
-}
-func fetchNetworksVirtualNetworkSubnets(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan<- interface{}) error {
-	vn := parent.Item.(network.VirtualNetwork)
-	if vn.Subnets == nil {
-		return nil
-	}
-	res <- *vn.Subnets
-	return nil
-}
 
-func fetchNetworksVirtualNetworkPeerings(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan<- interface{}) error {
-	vn := parent.Item.(network.VirtualNetwork)
-	if vn.VirtualNetworkPeerings == nil {
-		return nil
-	}
-	res <- *vn.VirtualNetworkPeerings
-	return nil
-}
+	response, err := svc.ListAll(ctx)
 
-func fetchNetworksVirtualNetworkGateways(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan<- interface{}) error {
```

Comment:
Extracted to `plugins/source/azure/resources/services/network/virtual_network_gateways.go`